### DISCUSSION TITLE
How to set node boundary conditions?

### DISCUSSION THREAD

**Discussion by PEI0214** [idaholab/moose#28410] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I want to set a boundary condition, I only have a set of nodes, is there any good way to do that?

**Comment by GiudGiud** [comment]
Hello

What kind of boundary condition? A flux? Or a dirichlet (set value)? Or a particular equation?
What type of variables are you using inside the domain?

**Reply by PEI0214** [reply]
Dirichlet boundary conditions. Nonlinear variable. Constrains the displacement of this node set.

**Reply by GiudGiud** [reply]
These should both be already doable on a node set. You specify the name or the id of the node set in the boundary parameter of the boundary condition

**Reply by PEI0214** [reply]
Only the number of the node is known, and the surface boundary is not defined. Is this also feasible? Like this.
![image](https://github.com/user-attachments/assets/a0376ff0-7e45-45ff-a0d4-d9b0bd3d98c5)

**Reply by GiudGiud** [reply, marked as answer]
You definitely need to create the node set in the mesh as well. 

If you have the ids, it should be easy to create the nodeset.
You can try the ExtraNodesetGenerator

https://mooseframework.inl.gov/source/meshgenerators/ExtraNodesetGenerator.html
